btree/btree.c

```diff
@@ -3740,6 +3761,61 @@ int64_t btree_open_tick(struct m0_sm_op *smop)
 	}
 }
 
+/**
+ * btree_close_tree_tick function is used to traverse through different states
+ * to facilitate the working of m0_btree_close().
+ *
+ * @param smop     represents the state machine operation
+ * @return int64_t returns the next state to be executed.
+ */
+int64_t btree_close_tree_tick(struct m0_sm_op *smop)
+{
+	struct m0_btree_op    *bop     = M0_AMB(bop, smop, bo_op);
+	struct td             *td_curr = bop->bo_arbor->t_desc;
+	struct nd             *nd_head = ndlist_tlist_head(&td_curr->
+							   t_active_nds);
+
+	switch (bop->bo_op.o_sm.sm_state) {
+	case P_INIT:
+		M0_ASSERT(td_curr->t_ref != 0);
+		if (td_curr->t_ref > 1)
+			tree_put(td_curr);
```

Comment:
For `t_ref > 1` case, the code should not falldown to P_TIMECHECK after tree_put();.

Reply:
Done.

Reply:
Do you want to set some error code in this case to indicate that tree is still open?

Reply:
No, I think so it wont be necessary as tree_put() is the primary purpose of for tree_close(). Indicating that it is open wont  be of any use.

Reply:
The primary purpose of tree_close is to close the tree. Ideally , tree_close should send the return code whether the operation was successful or not, so that the caller can take action. Anyways, if needed it can be added later.